server/app/src/main/java/com/google/heartrate/wearos/app/bluetooth/server/BluetoothAdvertiser.java

```diff
@@ -0,0 +1,87 @@
+package com.google.heartrate.wearos.app.bluetooth.server;
+
+import android.bluetooth.le.AdvertiseCallback;
+import android.bluetooth.le.AdvertiseData;
+import android.bluetooth.le.AdvertiseSettings;
+import android.bluetooth.le.BluetoothLeAdvertiser;
+import android.content.Context;
+import android.os.ParcelUuid;
+import android.util.Log;
+
+import com.google.heartrate.wearos.app.bluetooth.BluetoothUtils;
+import com.google.heartrate.wearos.app.gatt.GattException;
+
+import java.util.Set;
+import java.util.UUID;
+
+/**
+ * {@link BluetoothAdvertiser} is the wrapper over {@link BluetoothLeAdvertiser} class.
+ * <p>
+ * Advertising allows devices to broadcast information defining their intentions.
+ * Peripheral uses advertising to allow centrals to find it and, subsequently, to establish a connection with it.
+ * <p>
+ * A peripheral can use {@link BluetoothAdvertiser} to advertise its existence and show it's complete or partial
+ * list of GATT services it offers.
+ *
+ */
+public class BluetoothAdvertiser {
+    private static final String TAG = BluetoothAdvertiser.class.getSimpleName();
+
+    /** {@link BluetoothAdvertiser} for advertising process. */
+    private final BluetoothLeAdvertiser mBluetoothLeAdvertiser;
+
+    /** Advertising callback to receive async responses from {@link BluetoothAdvertiser} . */
+    final AdvertiseCallback mAdvertiseCallback = new AdvertiseCallback() {
+        @Override
+        public void onStartSuccess(AdvertiseSettings settingsInEffect) {
+            Log.i(TAG, "BluetoothLE Advertise Started.");
+        }
+
+        @Override
+        public void onStartFailure(int errorCode) {
+            Log.e(TAG, String.format("BluetoothLE Advertise Failed: %d",  + errorCode));
+        }
+    };
+
+    public BluetoothAdvertiser(Context context) throws GattException {
+        mBluetoothLeAdvertiser = BluetoothUtils.getBluetoothLeAdvertiser(context);
+    }
+
+    /**
+     * Start advertising centrals about peripheral existence.
+     *
+     * @param serviceUuids list of GATT services peripheral offers
+     */
+    public void start(Set<UUID> serviceUuids) {
```

Comment:
Maybe the name could be a bit more explicit like `startAdvertisingServices(...)`

Reply:
Renamed in [commit](https://github.com/googleinterns/heartrate-bt-wear/pull/5/commits/bca6eb03742aba84b0ea57c27e839fc22c06928a)